iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/ActivePowerControl.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.extensions;
+
+import com.powsybl.commons.extensions.AbstractExtension;
+import com.powsybl.iidm.network.Battery;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.Injection;
+
+/**
+ * @author Ghiles Abdellah <ghiles.abdellah at rte-france.com>
+ */
+public class ActivePowerControl<T extends Injection> extends AbstractExtension<T> {
```

Comment:
Is there a way to limit the allowed Injection subclasses? Maybe one day we'll add VSC converter stations?

Reply:
Without adding a new interface (for me that won't be a good idea), the only way I found is to put generic constructor (based on T) in protected and add 2 public constructors for generator and battery only. See code below.